src/ext/BalExtension/wixstdba/WixStandardBootstrapperApplication.cpp

```diff
@@ -576,6 +566,7 @@ class CWixStandardBootstrapperApplication : public CBalBaseBootstrapperApplicati
         __in HRESULT /*hrStatus*/
         )
     {
+        UpdateCacheProgress(100); // Make sure cache progress got set to 100%.
```

Comment:
Looks like I didn't consider when caching failed, probably should mark it 0%.  It doesn't really matter though, the progress bar would be hidden milliseconds later when going to the Failure page.